draft-ietf-mls-protocol.md

```diff
@@ -1477,14 +1477,25 @@ struct {
 } MLSMessageAuth;
 ~~~~~
 
-The `signature` field in an MLSMessageAuth object is computed using the signing
-private key corresponding to the public key, which was authenticated by the
-credential at the leaf of the tree indicated by the sender field. The signature
-is computed using `SignWithLabel` with label `"MLSMessageContentTBS"` and with a content
-that covers the message content and the wire format that will be used for this message.
-If the sender is a member of the group, the content also covers the
-GroupContext for the current epoch, so that signatures are specific to a given
-group and epoch.
+The signature is computed using `SignWithLabel` with label
+`"MLSMessageContentTBS"` and with a content that covers the message content and
+the wire format that will be used for this message. If the sender's
+`sender_type` is `Member`, the content also covers the GroupContext for the
+current epoch, so that signatures are specific to a given group and epoch.
+
+The sender MUST use the private key corresponding to the following signature key
+depending on the sender's `sender_type`:
+
+* `member`: The signature key contained in the Credential at the leaf with the
+  sender's `LeafNodeRef`
+* `external`: The signature key contained in the Credential at the index
+  indicated by the `sender_index` in the `external_senders` group context
+  extension (see Section {{external-senders-extension}}).
```

Comment:
```suggestion
* `external`: The signature key contained in the Credential at the index
  indicated by the `sender_index` in the `external_senders` group context
  extension (see Section {{external-senders-extension}}).  In this case, the
  `content_type` of the message MUST NOT be `commit`, since only members
  of the group or new joiners can send Commit messages.
```

Reply:
Could also be phrased as "MUST be only `proposal` / `application`.  Or could be added below these bullets.